src/Field/Field.tsx

```diff
@@ -0,0 +1,141 @@
+import React from 'react'
+import styled from 'styled-components'
+import { darken } from 'polished'
+
+import { Icon } from '../Icon'
+import { Text } from '../Text'
+import { Box } from '../Box'
+import { theme } from '../theme'
+
+type Props = {
+  children: React.ReactElement
+  showCaret?: boolean
+  renderAsTitle?: boolean
+  className?: string
+  id: string
+  error?: boolean // nb this needs to be optional for legacy support - but we will want to enforce in  the future
+  label?: string
+  outlined?: boolean
+  value: string
+  trailingIcon?: string
+  errorMsg?: string
+  dropdownKey?: string // nb only used by dropdown component but required on content in that context
+  required?: boolean
+  fullHeight?: boolean // used to make text areas full size
+}
```

Comment:
A big part of those props are shared between the inputs, we could create a common interface that can be used across them all? I'm thinking of `label`, `id`, `error`, `errorMsg`, `required`, `renderAsTitle`


```ts
type CommonFieldProps = {
  id?: string
  label?: string
  ...
}

type TextInputProps = { ... } & CommonFieldProps 
```

Reply:
I'm not sure I follow - what's the benefit of doing that, since all the types are encapsulated in the Field component?